src/Simplex_tree/include/gudhi/Simplex_tree.h

```diff
@@ -1667,6 +1667,41 @@ class Simplex_tree {
     return sh; // None of its faces has the same filtration.
   }
 
+ public:
+  /** \brief This function resets filtration value from a given dimension. Resets all the Simplex_tree when
+   * `min_dim = 0`.
+   * @param[in] filt_value The new filtration value.
+   * @param[in] min_dim The minimal dimension.
+   */
+  void reset_filtration(Filtration_value filt_value, int min_dim) {
+    for (auto& simplex : root_.members()) {
+      if (min_dim <= 0) {
+        simplex.second.assign_filtration(filt_value);
+      }
+      if (has_children(&simplex)) {
+        rec_reset_filtration(simplex.second.children(), filt_value, min_dim);
+      }
+    }
+    clear_filtration(); // Drop the cache.
+  }
+
+ private:
+  /** \brief Recursively resets filtration value from a given dimension.
+   * @param[in] sib Siblings to be parsed.
+   * @param[in] filt_value The new filtration value.
+   * @param[in] min_dim The maximal dimension.
+   */
+  void rec_reset_filtration(Siblings * sib, Filtration_value filt_value, int min_dim) {
+    for (auto sh = sib->members().begin(); sh != sib->members().end(); ++sh) {
+      if (min_dim <= dimension(sh)) {
```

Comment:
We shouldn't need to compute the dimension of each simplex, the dimension is the same for all the elements of this loop. I think the easiest would be to rename min_dim to min_depth (in the helper only), call recursively with `min_depth-1`, and compare min_depth to 0.

Reply:
Optimized on 723252b3